### DISCUSSION TITLE
Get the repetition ID from C++

### DISCUSSION THREAD

**Discussion by marsluca** [omnetpp/omnetpp#1194] [Q&A] [answered]
OMNeT++ allows the user to execute the same simulation multiple times with the parameter "repeat" in the INI file.
Is there a way to get the run ID in the simulation?

I saw there is a `getRunId()` function but I am not able to import [this class](https://github.com/omnetpp/omnetpp/blob/master/src/envir/resultfileutils.h). 
I tried the following import, but it doesn't work.

```
#include "envir/resultfileutils.h"
#include "resultfileutils.h"
#include "src/envir/resultfileutils.h"
#include "omnetpp/src/envir/resultfileutils.h" 
```

---- EDIT ----
I need the `repetitionID` not the `runID`, since run is more general and considers the different executions.

**Comment by avarga** [comment, marked as answer]
Here's the way to obtain the runId and the repetition:
```
cConfigurationEx *cfg = getEnvir()->getConfigEx();
std::string runId = cfg->getVariable(CFGVAR_RUNID);
int repetition = std::stoi(cfg->getVariable(CFGVAR_REPETITION));
```
